src/librustc_mir/const_eval.rs

```diff
@@ -110,34 +110,36 @@ fn op_to_const<'tcx>(
 fn eval_body_and_ecx<'a, 'mir, 'tcx>(
     tcx: TyCtxt<'a, 'tcx, 'tcx>,
     cid: GlobalId<'tcx>,
-    mir: Option<&'mir mir::Mir<'tcx>>,
```

Comment:
you can keep this argument (but without the option) and make the `&mir.promoted[index]` change at the call site.

If you want a sanity check, you can assert `cid.promoted.is_none() || mir.promoted.is_empty()` to make sure that callers always do the projection into promoted MIR

Reply:
There isn't an `InterpretCx` at the call site, so I'll have to move that too. At that point, there's only going to be a line or two of code left in `eval_body_and_ecx`. Do you want me to just remove the function then?

Reply:
yea, that function seems useless at this point, good idea